catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/TreUtility.java

```diff
@@ -38,6 +39,10 @@
     private static final FastDateFormat DATE_FORMATTER = FastDateFormat.getInstance(TRE_DATE_FORMAT,
             TimeZone.getTimeZone("GMT"));
 
+    private static Pattern FLOAT_PATTERN = Pattern.compile("([+-]?\\d*\\.\\d+)(?![-+0-9\\.])");
```

Comment:
This regex requires a . to be present in the string.  A float does not require a decimal value to be considered a float.  This regex will cause the convertToFloat method to return null if the evaluated Tre field is "3" when it should return 3.0.  Additionally, this regex doesn't consider exponent notation for floating point numbers.  Therefore, a value of "3e-3" would return null as well even though the correct and valid float value is 0.003.  Not sure if these are really options in a TRE field but a better regex to match a float would be: [-+]?\\d*\\.?\\d+([eE][-+]?\\d+)?

Reply:
Thanks, I'll use your regex instead.

Reply:
My regex for "convertToFloat" has been updated.